Fletcher32.cpp

```diff
@@ -19,25 +18,36 @@ Fletcher32::Fletcher32()
 
 void Fletcher32::begin(uint16_t s1, uint16_t s2)
 {
-  _s1    = s1;
-  _s2    = s2;
+  _s1    = (s1 == FLETCHER_32) ? 0 : s1;
+  _s2    = (s2 == FLETCHER_32) ? 0 : s2;
   _count = 0;
 }
 
 
 void Fletcher32::add(uint16_t value)
 {
   _count++;
+#ifdef ARDUINO_ARCH_AVR 
+  unsigned int t = _s1;
+  //  Serial.println("__builtin_uadd_overflow");
+  if (__builtin_uadd_overflow(t, value, &t)) {
+    t++;
+  }
+  _s1 = t;
+  t = _s2;
+  if (__builtin_uadd_overflow(t, _s1, &t)) {
+    t++;
+  } 
+  _s2 = t;
+#elif defined(ARDUINO_ARCH_SAMD) || defined(ESP32) || defined(ESP8266)
   _s1 += value;
-#if defined(ARDUINO_ARCH_SAMD) || defined(ESP32) || defined(ESP8266)
   _s1 = (_s1 & 65535UL) + (_s1 >> 16);
-#else
-  if (_s1 >= FLETCHER_32) _s1 -= FLETCHER_32;
-#endif
   _s2 += _s1;
-#if defined(ARDUINO_ARCH_SAMD) || defined(ESP32) || defined(ESP8266)
   _s2 = (_s2 & 65535UL) + (_s2 >> 16);
```

Comment:
Using `FLETCHER_32`?

Reply:
One of the "guidelines" in programming is that one uses no "magic numbers"
By naming them as a constant / #define the code becomes more self documenting (a little bit).
So yes, I would prefer FLETCHER32

Reply:
> (that would remove the need to do the modulo in the getFletcher() function.

Another night of sleep said "no no, not so".   The value can still become FLETCHERxx  
So still a test needed in **getFletcher()** 

if (_s2 == FLETCHERxx) _s2 = 0; 

A micro bit simpler than the subtraction. Might save a one or more instructions per call.

Reply:
Changed number constants to named constants.

Yes, `getFletcher()` is fine.

Storing the value first needs some time. Explicit casting could do the job here. But I think implicit casting is already OK.